RecoLocalFastTime/FTLCommonAlgos/plugins/BTLUncalibRecHitAlgo.cc

```diff
@@ -42,25 +42,53 @@ class BTLUncalibRecHitAlgo : public BTLUncalibratedRecHitAlgoBase {
 
 FTLUncalibratedRecHit 
 BTLUncalibRecHitAlgo::makeRecHit(const BTLDataFrame& dataFrame ) const { 
-  constexpr int iSample=2; //only in-time sample
-  const auto& sample = dataFrame.sample(iSample);
-  
-  double amplitude = double(sample.data()) * adcLSB_;
-  double time      = double(sample.toa()) * toaLSBToNS_;
 
-  // --- Correct the time for the time-walk and the constant delays
-  if ( amplitude > 0. )
-    time -= timeCorr_p0_*pow(amplitude,timeCorr_p1_) + timeCorr_p2_;
+  // The reconstructed amplitudes and times are saved in a std::pair
+  //    BTL tile geometry (1 SiPM): only the first value of the amplitude
+  //                                and time pairs is used.
+  //    BTL bar geometry (2 SiPMs): both values of the amplitude and
+  //                                time pairs are filled.
+
+  std::pair<float,float> amplitude(0.,0.);
+  std::pair<float,float> time(0.,0.);
 
   unsigned char flag = 0;
+
+  const auto& sampleLeft  = dataFrame.sample(0);
+  const auto& sampleRight = dataFrame.sample(1);
```

Comment:
is this guaranteed to exist?

Reply:
Yes. There are always two samples in the DataFrame.